cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -294,3 +355,327 @@ printStakeDistribution (PoolDistr stakeDist) = do
 -- TODO: we could show the VRF id, but it will then not fit in 80 cols
 --      , show vrfKeyId
         ]
+
+
+-- From Cardano.Api
+
+-- | Query the UTxO, filtered by a given set of addresses, from a Shelley node
+-- via the local state query protocol.
+--
+-- This one is Shelley-specific because the query is Shelley-specific.
+--
+queryUTxOFromLocalState
+  :: Network
+  -> SocketPath
+  -> QueryFilter
+  -> Point (ShelleyBlock TPraosStandardCrypto)
+  -> ExceptT LocalStateQueryError IO (Ledger.UTxO TPraosStandardCrypto)
+queryUTxOFromLocalState network socketPath qFilter point = do
+  utxoFilter <- hoistEither $ applyUTxOFilter qFilter
+  let pointAndQuery = (point, utxoFilter)
+  newExceptT $ queryNodeLocalState
+    nullTracer
+    (pClientInfoCodecConfig . protocolClientInfo $ mkNodeClientProtocolShelley)
+    network
+    socketPath
+    pointAndQuery
+
+applyUTxOFilter
+  :: (blk ~ ShelleyBlock TPraosStandardCrypto, c ~ TPraosStandardCrypto)
+  => QueryFilter
+  -> Either LocalStateQueryError (Query blk (Ledger.UTxO c))
+applyUTxOFilter qFilter =
+    case qFilter of
+      FilterByAddress addrs -> do shelleyAddrs <- checkAddresses $ partitionAddresses addrs
+                                  Right $ GetFilteredUTxO shelleyAddrs
+      NoFilter -> Right GetUTxO
+  where
+    checkAddresses :: (Set ByronAddress, Set ShelleyAddress) -> Either LocalStateQueryError (Set ShelleyAddress)
+    checkAddresses (byronAddrs, shelleyAddrs) = if Set.null byronAddrs
+                                                then Right $ shelleyAddrs
+                                                else Left $ ByronAddressesNotSupportedError byronAddrs
```

Comment:
All this checking and filtering can go away if we switch the `QueryFilter` over to use the new api for its address type. We just need to use `Address Shelley` and then the conversion function
```
toShelleyAddr :: Address era -> Shelley.Addr ShelleyCrypto
toShelleyAddr (ByronAddress addr)        = Shelley.AddrBootstrap
                                             (Shelley.BootstrapAddress addr)
toShelleyAddr (ShelleyAddress nw pc scr) = Shelley.Addr nw pc scr
```